src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -69,8 +116,12 @@ public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterf
      * @param string       $relationAlias           the previous relation alias to keep it unique
      * @param bool         $wasLeftJoin             if the relation containing the new one had a left join, we have to force the new one to left join too
      */
-    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass, array $propertyMetadataOptions = [], string $originAlias = 'o', string &$relationAlias = 'a', bool $wasLeftJoin = false)
+    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass, array $propertyMetadataOptions = [], string $originAlias = 'o', string &$relationAlias = 'a', bool $wasLeftJoin = false, $depth = 0)
     {
+        if ($depth > $this->maxJoins) {
+            throw new \RuntimeException('Too many joined relations, might be an infinite loop! Check out serialization groups!');
```

Comment:
`ApiPlatform\Core\Exception\RuntimeException` instead? You have to add it in the PHPDoc.